bioblend/galaxyclient.py

```diff
@@ -54,11 +55,18 @@ def __init__(
             for scheme in ("https://", "http://"):
                 log.warning(f"Missing scheme in url, trying with {scheme}")
                 with contextlib.suppress(requests.RequestException):
-                    r = requests.get(
-                        scheme + url,
-                        timeout=self.timeout,
-                        verify=self.verify,
-                    )
+                    if session is not None:
+                        r = session.get(
+                            scheme + url,
+                            timeout=self.timeout,
+                            verify=self.verify,
+                        )
+                    else:
+                        r = requests.get(
+                            scheme + url,
+                            timeout=self.timeout,
+                            verify=self.verify,
+                        )
```

Comment:
Instead of duplicating all the get/post/put/delete calls, can you do something like this at line 50:
```
        if session is None:
            session = requests.Session()
        self.session = session
```

and then use `self.session` everywhere?

Reply:
Thanks! Cleans up the code!